feat(RTLplay): revamp 2.0.0 + features added

pierrequiroul opened the pull request:
## Description 
<!-- A clear and detailed description of the changes, referencing issues if applicable -->
Added features :
- now displays radio bel rtl (from the same tv group)
- now correctly displays the new RTL district channel
- getThumbnail updated to give images a better gradient border
- new icons (including vinyl for music)
- changes to the way information is displayed (formatting)
- slideshow on media page between poster and background image

Changes/fixes/qol:
- displays new timestamps correctly
- moved utilities functions to a separate file
- renamed presence parameters to make them clearer to understand (avoid typos by using them, especially for me lol)
- reorganized tags to display the 3 most relevant ones first (for the visual of the new extension)
- added functions getLocalizedAssets (for the ad icon), limitText (largeImageText has a limit of 128 characters)
- added crop presets for the getThumbnail function (to modify values in a single location)
- fixed the search page

So, with all that, I modified the version from 1.0.3 to 2.0.0

Previous look:
https://github.com/PreMiD/Presences/pull/8605

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<!-- 
    Screenshots of the presence settings (if applicable) and at least TWO screenshots of the presence displaying correctly
    Including these screenshots will assist the reviewing processes thus speeding up the process of the pull request being merged
-->

[https://imgur.com/a/vFVSzog](https://imgur.com/a/vFVSzog)

Home page:
![https://i.imgur.com/djEaucP.png](https://i.imgur.com/djEaucP.png)
Search page:
![yEIbEEg](https://i.imgur.com/yEIbEEg.png)
![1D4vrsb](https://i.imgur.com/1D4vrsb.png)
Media page
![2BOqCcI](https://i.imgur.com/qwEgowh.png)
![74PHFUS](https://i.imgur.com/2BOqCcI.gif)
![qwEgowh](https://i.imgur.com/74PHFUS.png)
Watching a TV Show (usePresenceName setting)
![PMhPaGN](https://i.imgur.com/PMhPaGN.png)
![6l7SGZc](https://i.imgur.com/6l7SGZc.png)
Watching a livestream
![T8hs6Ma](https://i.imgur.com/T8hs6Ma.png)
![7dim9PG](https://i.imgur.com/7dim9PG.png)
Radio livestream (music clip channel)
![cC9zaYl](https://i.imgur.com/cC9zaYl.gif)
Regular radio player (bel rtl and radio contact websites)
![4TKARwR](https://i.imgur.com/4TKARwR.gif)
![6jrzBZO](https://i.imgur.com/6jrzBZO.gif)
![https://i.imgur.com/su6p58R.png](https://i.imgur.com/su6p58R.png)
</details>

Bas950 commented on websites/R/RTLplay/util.ts:
```diff
@@ -0,0 +1,325 @@
+	return document.querySelector(selector) !== null;
+}
+
+// Copy of the function in Youtube utils
+let cachedTime = 0;
+export function adjustTimeError(time: number, acceptableError: number): number {
+	if (Math.abs(time - cachedTime) > acceptableError) cachedTime = time;
+	return cachedTime;
+}
```
this shouldn't be needed anymore.

pierrequiroul replied:
Hi @Bas950 ! 
I wanted to add the features keeping in mind that it works correctly on older versions too. I noticed that with Version: 2.7.0 (ba99913), when a media is played, the extension spammed headless_sessions in the service worker because the start timestamp was updated too fast, resulting in a lot of headless_sessions failing in rate limited. This meant that the change to other pages was much less responsive or was blocking or taking a long time because I guess there's probably a queue set up for status discord changes. So I looked at how other presences handled timestamps, found this function and it solved the problem. The headless_sessions stabilize after a while and only send when necessary, and the page change is more responsive.

I also tested with the latest recoded build Version: 2.7.0 (e9ff721) if this problem existed and it doesn't. So I agree with you and I came to the conclusion that this function was temporary and could be removed in the future. During development I went back and forth between the 2 versions to see if everything worked as I imagined it would in both.

With adjust function
![2 7 0_adjust](https://github.com/user-attachments/assets/69a0b0c9-dcdc-4778-93ad-cd4428a56802)
Without
![2 7 0_noadjust](https://github.com/user-attachments/assets/2c07ce4f-89cb-4672-a63c-386ae17079aa)

Bas950 replied:
@Timeraa might be able to confirm, but I don't think we will release these presences to the current bundled extension version. So it will be released with the new alpha/beta version.

Timeraa replied:
I'm not quite sure, I hope to finish 2.7.0 this year but no promises there but the spamming issue still somewhat happens with that function on 2.6.x because of a broken if check

pierrequiroul commented:
@Bas950 @theusaf, I removed the adjustTimeError function as requested. 
I also added altnames and new tags in the metadata.json since your last reviews
